src/main/java/pl/pandziak/ticketbookingapp/api/ScreeningEndpoint.java

```diff
@@ -0,0 +1,47 @@
+package pl.pandziak.ticketbookingapp.api;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import pl.pandziak.ticketbookingapp.domain.screening.ScreeningFacade;
+import pl.pandziak.ticketbookingapp.domain.screening.dtos.ScreeningResponseDto;
+import pl.pandziak.ticketbookingapp.domain.screening.dtos.ScreeningsRequestDto;
+import pl.pandziak.ticketbookingapp.domain.screening.dtos.ScreeningsResponseDto;
+import pl.pandziak.ticketbookingapp.domain.screening.exceptions.InvalidDateFormatException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+@RestController
+@RequestMapping("/screenings")
+class ScreeningEndpoint {
+
+    private final ScreeningFacade screeningFacade;
+
+    ScreeningEndpoint(ScreeningFacade screeningFacade) {
+        this.screeningFacade = screeningFacade;
+    }
+
+    @GetMapping("/{id}")
+    ScreeningResponseDto getScreening(@PathVariable String id) {
+        return screeningFacade.getScreening(id);
+    }
+
+    @GetMapping
+    ScreeningsResponseDto getScreenings(@RequestParam("from") String from) {
+        LocalDateTime parsedFrom = parseDate(from);
+        return screeningFacade.getScreeningsForDayAfterGivenHour(new ScreeningsRequestDto(parsedFrom));
```

Comment:
Czemu to opakowanie w DTO?

Reply:
Opakowanie w DTO zostało zastosowane z myślą o rozszerzalności zapytania. Jeżeli teraz postanowiłabym zmienić założenie, że zamiast od podanej godziny do końca dnia wyświetlam filmy między dwiema datami podanymi przez użytkownika - nie byłoby potrzeby zmienić przekazywanego obiektu do fasady. Dodane zostało by drugie pole w istniejącym już DTOsie.

Reply:
Tzn zgadzam się z Tobą, chciałem tylko potwierdzić uzasadnienie ;)